OrbitGl/CallStackDataView.cpp

```diff
@@ -187,6 +210,30 @@ void CallStackDataView::OnDataChanged() {
   DataView::OnDataChanged();
 }
 
+void CallStackDataView::SetFunctionsToHighlight(std::vector<uint64_t> sampling_functions) {
+  functions_to_highlight_.resize(indices_.size());
+  std::fill(functions_to_highlight_.begin(), functions_to_highlight_.end(), false);
+  for (uint64_t function_address : sampling_functions) {
+    std::optional<int> index_in_callstack =
+        GetHighlightedIndexFromFunctionAddress(function_address);
+    if (index_in_callstack.has_value()) {
+      functions_to_highlight_[index_in_callstack.value()] = true;
+    }
+  }
+}
```

Comment:
As discussed offline: I didn't think about the fact that the `FunctionInfo` may be empty if this function comes from a module that is not loaded, and that the addresses displayed in the Callstack view are not function base addresses, but the actual sampling addresses. I can follow your reasoning for using a vector here to make the lookup by index as fast as possible.

I still think this is quite hard to dig through - @pierricgimmig would be great if the three of us could find some time to discuss this!

Reply:
Got suggestion from @florian-kuebler: For function samples in the Callstack view, we can compute the *function base address* as follows:
- based on `FunctionInfo`  (See `FindFunctionByAddress`) if `FunctionInfo` is not `nullptr`, 
- otherwise, based on the `LinuxAddressInfo` (see `GetFunctionByName`), where the function address would be ``the absolute address - the offset in function".

In the current code base, we have `SamplingDataPostProcessor::MapAddressToFunction`, which does exactly this. Thanks @florian-kuebler for adding the method `CaptureData::GetFunctionBaseAddressByAddress`  to get the function base address in [PR#1641](https://github.com/google/orbit/pull/1641/files).